library/lib.ml

```diff
@@ -328,16 +335,17 @@ let start_compilation s mp =
   lib_state := { !lib_state with comp_name = Some s;
                                  path_prefix = prefix }
 
+let open_blocks_message es =
+  let open_block_name = function
+      | oname, OpenedSection _ -> str "section " ++ pr_id (basename (fst oname))
+      | oname, OpenedModule (ty,_,_,_) -> str (module_kind ty) ++ spc () ++ pr_id (basename (fst oname))
+      | _ -> assert false in
+  str "The " ++ pr_enum open_block_name es ++ str " need to be closed."
```

Comment:
I implemented this with a simple check on the length.

We shouldn't have a generic function. Here's a pluralizer for English: https://github.com/clips/pattern/blob/master/pattern/text/en/inflect.py#L107-L443. It's 300 lines of Python and achieves 95% accuracy.